src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKeyBlobFormat.cs

```diff
@@ -0,0 +1,197 @@
+    ///     require or return a key blob format take the name as a string, we use this string wrapper class to
+    ///     specifically mark which parameters and return values are expected to be key blob formats.  We also
+    ///     provide a list of well known blob formats, which helps Intellisense users find a set of good blob
+    ///     formats to use.
+    /// </summary>
+    public sealed class CngKeyBlobFormat : IEquatable<CngKeyBlobFormat>
+    {
+        public CngKeyBlobFormat(string format)
+        {
+            Contract.Ensures(!string.IsNullOrEmpty(_format));
+
+            if (format == null)
+                throw new ArgumentNullException("format");
+            if (format.Length == 0)
+                throw new ArgumentException(SR.Format(SR.Cryptography_InvalidKeyBlobFormat, format), "format");
+
+            _format = format;
+        }
+
+        /// <summary>
+        ///     Name of the blob format
+        /// </summary>
+        public string Format
+        {
+            get
+            {
+                Contract.Ensures(!string.IsNullOrEmpty(Contract.Result<string>()));
+                return _format;
+            }
+        }
+
+        public static bool operator ==(CngKeyBlobFormat left, CngKeyBlobFormat right)
+        {
+            if (object.ReferenceEquals(left, null))
+            {
+                return object.ReferenceEquals(right, null);
+            }
+
+            return left.Equals(right);
+        }
+
+        [Pure]
+        public static bool operator !=(CngKeyBlobFormat left, CngKeyBlobFormat right)
+        {
+            if (object.ReferenceEquals(left, null))
+            {
+                return !object.ReferenceEquals(right, null);
+            }
+
+            return !left.Equals(right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            Contract.Assert(_format != null);
+
+            return Equals(obj as CngKeyBlobFormat);
+        }
+
+        public bool Equals(CngKeyBlobFormat other)
+        {
+            if (object.ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return _format.Equals(other.Format);
+        }
+
+        public override int GetHashCode()
+        {
+            Contract.Assert(_format != null);
+            return _format.GetHashCode();
+        }
+
+        public override string ToString()
+        {
+            Contract.Assert(_format != null);
+            return _format;
+        }
+
+        //
+        // Well known key blob formats
+        //
+
+        public static CngKeyBlobFormat EccPrivateBlob
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<CngKeyBlobFormat>() != null);
+
+                if (s_eccPrivate == null)
+                {
+                    s_eccPrivate = new CngKeyBlobFormat("ECCPRIVATEBLOB");      // BCRYPT_ECCPRIVATE_BLOB
+                }
+
+                return s_eccPrivate;
```

Comment:
FWIW, even if you don't factor this out into a helper, I personally like the more concise:
```C#
return s_eccPrivate ?? (s_eccPrivate = new CngKeyBlobFormat("ECCPRIVATEBLOB"));
```
I realize it's a matter of taste.  Just sharing :)